### PR TITLE
Add Vale linting for our docs

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

- I noticed that another open source project had taken the Homebrew style guide and made it into [Vale linting rules](https://github.com/testthedocs/vale-styles/tree/master/Homebrew). This copies them into here, so that we can make use of them. Thanks!
- What use is a style guide if our own docs don't at least try to adhere to it? This aims to make the rules more visible to all contributors.
- In order for these to do anything, you'll have to `brew install vale` and run `vale` in the root of this repo. It will look for Markdown files.
- A next step would be adding this as a pre-commit hook, or...
- The GitHub Action would have been really good, but [it doesn't support forks](https://github.com/errata-ai/vale-action#limitations).

-----

Current status with `vale .`:

```
✖ 226 errors, 851 warnings and 0 suggestions in 233 files.
```

Current status with `vale docs/`:

```
✖ 15 errors, 89 warnings and 0 suggestions in 45 files.
```

There are probably some false positives, and some legitimate exceptions, but it's worth looking at.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
Nice! I'm strongly in favour of this.

**Comment by issyl0** [comment]
Having looked at the remaining linting errors further, I would like to propose a revision to the Prose Style Guide that would resolve a big portion of the linting errors:

Changing or removing the "Sentence case in section headings, not Title Case" guideline.

Correcting all the errors would be a lot of work, and title case makes sense for... Well, titles and section headings. Seems like we use title case for pretty much every Markdown section heading other than those in the style guide itself.

Opinions welcome! This PR is draft for a reason.

**Comment by issyl0** [comment]
The impact of removing the Title Case rule (and lint):

```
✖ 7 errors, 1 warning and 0 suggestions in 45 files.
```

**Comment by EricFromCanada** [comment]
What if we change it to sentence case for `<h3>` and below with `heading.h3`?

File: docs/Taps.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ dunn/emacs
 ```
 
 * `brew tap <user/repo>` makes a shallow clone of the repository at
-  https://github.com/user/repo. After that, `brew` will be able to work on
+  https://github.com/user/repository. After that, `brew` will be able to work on
```
Comment (jonchang) [inline docs/Taps.md]: This should be left as-is

File: docs/vale-styles/Homebrew/Terms.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+extends: substitution
+message: Use '%s' instead of '%s'.
+link: 'https://github.com/Homebrew/brew/blob/master/docs/Prose-Style-Guidelines.md#terminology-words-and-word-styling'
```
Comment (EricFromCanada) [inline docs/vale-styles/Homebrew/Terms.yml]: Add `scope: paragraph` to clear those last few errors.

File: docs/Taps.md (hunk 1 above)
Comment (issyl0) [reply docs/Taps.md]: I thought so too. The robots can't rule us all the time! I'll work on this later.

**Comment by MikeMcQuaid** [comment]
> Having looked at the remaining linting errors further, I would like to propose a revision to the Prose Style Guide that would resolve a big portion of the linting errors:
> 
> Changing or removing the "Sentence case in section headings, not Title Case" guideline.

I'm game for that providing we (can?) lint the "title case" case?

**Comment by MikeMcQuaid** [comment]
Bonus points if this can be convinced to somehow lint `formula.rb` comments and https://github.com/homebrew/brew.sh (in a separate/later PR).

**Comment by issyl0** [comment]
Thanks for the feedback. I've got the number of linting problems down to 0 for the `docs/` directory, and have only implementing Title Case for h1 headings. I plan to push the changes we made to the linter configs back [upstream](https://github.com/testthedocs/vale-styles). ♻️

For "how to keep ourselves in line with our own style guide" going forward, what are people's thoughts?

A) Wait for the GitHub Action to support forks, or
B) Implement a pre-commit hook?

There's a barrier to entry with a pre-commit hook (getting people to install them is difficult), and Homebrew doesn't have any prior art for them. We seem to be trying to converge everything else on GitHub Actions, so I'm tempted to wait it out.

File: docs/Taps.md [hunk 3]
```diff
@@ -16,12 +16,14 @@ mistydemeo/tigerbrew
 dunn/emacs
 ```
 
+<!-- vale Homebrew.Terms = OFF -->
```
Comment (MikeMcQuaid) [inline docs/Taps.md]: What's this doing?

**Comment by MikeMcQuaid** [comment]
> For "how to keep ourselves in line with our own style guide" going forward, what are people's thoughts?

Stupid question but how/where are these checks actually running? Is it an open source project we can run as part of our own build?

File: docs/Taps.md (hunk 3 above)
Comment (issyl0) [reply docs/Taps.md]: Filtering out the one "repo" we want to keep as "repo" and not force a change to "repository". It's like the Rubocop disable lines.

**Comment by issyl0** [comment]
> > For "how to keep ourselves in line with our own style guide" going forward, what are people's thoughts?
> 
> Stupid question but how/where are these checks actually running? Is it an open source project we can run as part of our own build?

Yep, it's a CLI tool. The source is at [errata-ai/vale](https://github.com/errata-ai/vale). I'll look at adding this as part of Homebrew/brew CI.

**Comment by MikeMcQuaid** [comment]
> Yep, it's a CLI tool. The source is at [errata-ai/vale](https://github.com/errata-ai/vale). I'll look at adding this as part of Homebrew/brew CI.

👍 thanks. I think if that's done it should be able to handle forks?

**Comment by issyl0** [comment]
I've not forgotten about this. I'm off work for Christmas from this Wednesday, so I should have more time after that!

**Comment by issyl0** [comment]
The `vale` lint CI check works! The only remaining thing is the man page generation which seems to lowercase all Title Cased section headings in `docs/Manpage.md`. Should we exclude docs/Manpage.md from the checks, or change the man page generation script to not do this? I'm leaning towards the latter for consistency with our rules.

**Comment by MikeMcQuaid** [comment]
> I'm leaning towards the latter for consistency with the above discussions on the subject of title casing.

Yes, that would be nice!

**Comment by issyl0** [comment]
All done.

I've re-requested a review given the number of added commits since the original approval a week ago. ⏳

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -105,6 +105,12 @@ jobs:
         # These cannot be queried at the macOS level on GitHub Actions.
         HOMEBREW_LANGUAGES: en-GB
 
+    - name: Run vale for docs linting
```
Comment (MikeMcQuaid) [inline .github/workflows/tests.yml]: Can this run after `brew readall`? Thanks!

File: docs/Manpage.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
-brew(1) -- The missing package manager for macOS
+brew(1) -- The Missing Package Manager for macOS
```
Comment (MikeMcQuaid) [inline docs/Manpage.md]: Would be good to update Homebrew/brew.sh accordingly too.

File: docs/Taps.md (hunk 3 above)
Comment (MikeMcQuaid) [reply docs/Taps.md]: Cool, would be nice to document each of these cases as to why it's being disabled?

File: docs/Manpage.md (hunk 5 above)
Comment (issyl0) [reply docs/Manpage.md]: https://github.com/Homebrew/brew.sh/pull/436

**Comment by MikeMcQuaid** [comment]
Woohoo, thanks @issyl0!

**Pull request merged**